src/main/java/ru/practicum/shareit/item/storage/InMemoryItemStorage.java

```diff
@@ -0,0 +1,60 @@
+package ru.practicum.shareit.item.storage;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import ru.practicum.shareit.IdCreator;
+import ru.practicum.shareit.exception.ResourceException;
+import ru.practicum.shareit.item.dto.ItemDto;
+import ru.practicum.shareit.item.mapping.ItemMapper;
+import ru.practicum.shareit.item.model.Item;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class InMemoryItemStorage implements ItemStorage {
+
+    private final IdCreator idCreator = new IdCreator();
+    private final HashMap<Integer, Item> items = new HashMap<>();
```

Comment:
Принято объявлять объект с интерфейсом, а создавать с определенной реализацией. Данный подход делает программу более поддерживаемой и не зависящей от определенной реализации
```suggestion
    private final Map<Integer, Item> items = new HashMap<>();
```